### ISSUE TITLE
CSS format: Disallow using of shorthand

### ISSUE THREAD

**Issue by marcoscv-work** [liferay/liferay-npm-tools#221]
Disallow using of [shorthand](https://developer.mozilla.org/en-US/docs/Web/CSS/Shorthand_properties)

Examples of **correct** code:

```
.example {
   padding-top: 5px;
   padding-right: 2px;
}
```

Examples of **incorrect** code:

```
.example {
   padding: 5px 2px 0 0;
}
```

**Comment by pat270** [comment]
I don't think this is a good idea to enforce across the board. This is done in Clay CSS because variables are being heavily leveraged. It gives more customization options.

**Comment by marcoscv-work** [comment]
It should be a rule to be used in our development, not for customers.
Could we show an alert or something not so intrusive?

**Comment by jbalsas** [comment]
On top of the rule, we should add a rationale, an explanation as to why we'd like to enforce this. That way we can better evaluate if this makes sense as a guideline and something to enforce or not.

@marcoscv-work, @eduardoallegrini, could you also add in the initial comment the **why**?

**Comment by marcoscv-work** [comment]
I totally agree with some of the last issues could be closer to "recommendations" than "be forced to" or "ensure".
So if you guys think some of them should be only a recommendation the alert message could be only a recommendation or advise

**Comment by jbalsas** [comment]
One other thing, on a CI build, it's hard to tell apart a warning from an error, so when I say "recommendations" I mean written recommendations in our https://github.com/liferay/liferay-frontend-guidelines repo and not so much a build-time rule (although we could build it and just not automate it).

**Comment by marcoscv-work** [comment]
If do we all agree, we could simply close the issue

**Comment by eduardoallegrini** [comment]
I did some "google research" and I found something like:
- "don't use shorthand to avoid setting not used attributes" (in the case of margin, padding)
- "don't use shorthand to avoid to overwrite values by mistake" (in the case of font, background)

These reasons are ok but they don't seem really important, so I agree to leave them as a simple recommendation in the guideline

**Comment by pat270** [comment]
+1 for written recommendation

**Comment by wincent** [comment]
Just cross-referencing [this issue](https://github.com/liferay/liferay-npm-tools/issues/344) here. As described in [this comment](https://github.com/liferay/liferay-npm-tools/issues/344#issuecomment-580739620), we now have style-lint [active in liferay-portal](https://github.com/brianchandotcom/liferay-portal/pull/83654), and [something like 49 rules are turned on](https://github.com/liferay/liferay-npm-tools/blob/649f5d0b9d3af3dd003670d4b2e8411cb61a5dca/packages/liferay-npm-scripts/src/config/stylelint.json) and additionally, we have some [checking facilities that operate outside of ESLint/style-lint](https://github.com/liferay/liferay-npm-tools/blob/649f5d0b9d3af3dd003670d4b2e8411cb61a5dca/packages/liferay-npm-scripts/src/scripts/check/preflight.js) and [an example PR](https://github.com/liferay/liferay-npm-tools/pull/348) showing how we could use this to do things like enforce file-naming conventions.

This issue was one of the ones listed as a possible action item, but in the end we didn't get to it because isn't clear that we have consensus on this, and it may be that a non-technical (ie. not automatically enforced) solution — perhaps just documentation, which could maybe go in the [liferay-frontend-guidelines](https://github.com/liferay/liferay-frontend-guidelines/).
